### PR TITLE
Add AWS-specific support to SDK

### PR DESCRIPTION
**Pull request by kbrockhoff**
Provides Resource label population using the EC2 metadata endpoint. Provides IDs generator compatible with AWS X-Ray.

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/resources/ResourceConstants.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+package io.opentelemetry.sdk.resources;
+
+/**
+ * Provides constants for resource semantic conventions defined by the OpenTelemetry specification.
+ *
+ * @see <a
+ *     href="https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md">Resource
+ *     Conventions</a>
+ */
+public class ResourceConstants {
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/resources/ResourceConstants.java]: Please split this in a separate PR :)

Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/resources/ResourceConstants.java]: Also we should consider if using am xml, yaml file for all the constants is appropriated and share that between libraries.

Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/resources/ResourceConstants.java]: should these semantic conventions be encoded in some protobuf definitions, so they can be shared across all the language implementations?

Comment (kbrockhoff) [reply sdk/src/main/java/io/opentelemetry/sdk/resources/ResourceConstants.java]: I checked the projects. opentelemetry-go, opentelemetry-collector, opentelemetry-dotnet, opentelemetry-auto-instr-java, opentelemetry-js all have constants for semantic conventions defined already. I can see the benefit of having one file to update but not sure it is workable.

I will put the constants in a separate PR. Should I put them in the API where they are available to users? It seems to make sense given they are part of the specification.

Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/resources/ResourceConstants.java]: They are part of the SDK specification (resource is part of the SDK). The attributes for trace are part of the API and for those we probably need them in the API.

**Comment by kbrockhoff** [comment]
Will resubmit after resource label constants are merged.
